libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -23,6 +24,22 @@ TransactionManager::TransactionManager(const FullNodeConfig &conf, std::shared_p
   }
 }
 
+uint64_t TransactionManager::estimateTransactions(const SharedTransactions &trxs,
+                                                  std::optional<PbftPeriod> proposal_period) const {
+  util::ThreadPool estimation_thread_pool_(std::thread::hardware_concurrency() / 2);
```

Comment:
this is declaring new threadpool everytime it enters estimateTransactions function ?

Reply:
yeah, because there is a possibility of multiple dag blocks verifications(and this estimations) at the same time

Reply:
hmm and why is it also DagManager class member ?

Reply:
I will rework this using this https://github.com/Taraxa-project/taraxa-node/pull/2998